### PR TITLE
Add support for automatic title & icon modification

### PR DESCRIPTION
**Pull request by palinek**
Another implementation for  #52

### CODE REVIEW

**Comment by yan12125** [comment]
An idea after trying this branch: when a new tab is opened and no title-setting command is issued, the tab text is an empty string. I guess keep the old behavior "Shell no. XX" is better.

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -740,3 +738,16 @@ void MainWindow::addNewTab()
...
+{
+    QString title;
+    QIcon icon;
+    if (-1 != index)
+    {
+        title = consoleTabulator->tabText(index);
+        icon = consoleTabulator->tabIcon(index);
+    }
+    setWindowTitle(title.isEmpty() ? QStringLiteral("QTerminal") : title);
+    setWindowIcon(icon.isNull() ? QIcon::fromTheme("utilities-terminal") : icon);
```
Comment (yan12125) [inline src/mainwindow.cpp]: Maybe this line should be removed. For example if I run ```mpv xxx.mp4``` from QTerminal, the window icon becomes the same as mpv's. It brings inconvenience in LXQT, as I use icons in the Alt-Tab window to tell different applications.

Comment (palinek) [inline src/mainwindow.cpp]: So you propose, not to change the qterminal's main icon based on icon of selected terminal? Sounds reasonable... but IMO reasonable is it the other way round. Maybe we should make it configurable. What do you mean?

Comment (yan12125) [inline src/mainwindow.cpp]: > So you propose, not to change the qterminal's main icon based on icon of selected terminal?

Yes. And a configurable feature sounds a better option than disabling it.

**Comment by palinek** [comment]
> An idea after trying this branch: when a new tab is opened and no title-setting command is issued, the tab text is an empty string. I guess keep the old behavior "Shell no. XX" is better.

Please, try with the @aff2562 (needs also update of the lxde/qtermwidget#84)

**Comment by yan12125** [comment]
Thanks. Now looks all good.

**Comment by palinek** [comment]
@yan12125 Now also the configuration for changing window title/icon should be done.

**Comment by yan12125** [comment]
Configuration options regarding window title and icon work fine. Many thanks!

**Comment by pmattern** [comment]
Works like a charm here.

GTM.

**Pull request merged**